data_structures/stack/main.c

```diff
@@ -19,15 +19,16 @@ int main()
     //function for poping the element out
     void pop()
     {
-        if(top==0)
+        if(top==-1)
         {
             printf("\nstack is empty");
         }
         else
         {
             int item;
-            item=a[top];
-            top=top-1;
+            
+            a[top]=item;
```

Comment:
This is a logical error. Because you do not assign a value to the variable ```item```. After that you print it...